pkg/controller/infinispan/resources/config/xsite.go

```diff
@@ -20,51 +20,51 @@ import (
 )
 
 // ComputeXSite compute the xsite struct for cross site function
-func ComputeXSite(infinispan *ispnv1.Infinispan, kubernetes *kube.Kubernetes, service *corev1.Service, logger logr.Logger, xsite *config.XSite) error {
-	if infinispan.HasSites() {
-		siteServiceName := infinispan.GetSiteServiceName()
-		localSiteHost, localSitePort, err := getCrossSiteServiceHostPort(service, kubernetes, logger)
-		if err != nil {
-			logger.Error(err, "error retrieving local x-site service information")
-			return err
-		}
+func ComputeXSite(infinispan *ispnv1.Infinispan, kubernetes *kube.Kubernetes, service *corev1.Service, logger logr.Logger) (error, *config.XSite) {
+	siteServiceName := infinispan.GetSiteServiceName()
+	localSiteHost, localSitePort, err := getCrossSiteServiceHostPort(service, kubernetes, logger)
+	if err != nil {
+		logger.Error(err, "error retrieving local x-site service information")
+		return err, nil
+	}
+
+	if localSiteHost == "" {
+		msg := "local x-site service host not yet available"
+		logger.Info(msg)
+		return fmt.Errorf(msg), nil
+	}
+
+	logger.Info("local site service",
+		"service name", siteServiceName,
+		"host", localSiteHost,
+		"port", localSitePort,
+	)
+
+	xsite := &config.XSite{
+		Address: localSiteHost,
+		Name:    infinispan.Spec.Service.Sites.Local.Name,
+		Port:    localSitePort,
+	}
 
-		if localSiteHost == "" {
-			msg := "local x-site service host not yet available"
+	remoteLocations := findRemoteLocations(xsite.Name, infinispan)
+	for _, remoteLocation := range remoteLocations {
+		var err error
+		if remoteLocation.Host != "" {
+			err = appendBackupSite(&remoteLocation, xsite)
+		} else if remoteLocation.URL != "" { // lookup remote service via kubernetes api
+			err = appendRemoteLocation(infinispan, &remoteLocation, kubernetes, logger, xsite)
+		} else {
+			msg := fmt.Sprintf("invalid xsite location, remote name: %s does not define host or url", remoteLocation.Name)
 			logger.Info(msg)
-			return fmt.Errorf(msg)
+			err = fmt.Errorf(msg)
```

Comment:
Redundant info logging, we can just return an updated error instance and let `Process()` handle it.